### PR TITLE
Migrate Customer list actions

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrates all bulk actions and _Enable_, _Newsletter_, _Partner offers_ and _Delete_ row actions.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10559
| How to test?  | Migrated actions should be working now. Check `admin/sell/customers` to access this page. **NOTE: build assets before testing**. Travis report is red because of conflicts but this PR can be validated 👍 .

customers_grid

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11439)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js:29]: can't we make this extension more configurable and so on "reusable" ?

File: admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js:29]: can't we make this extension more configurable and so on reusable?

File: src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php
Comment (mickaelandrieu) [inline src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php]: shouldn't be this condition instead? I don't understand the error message linked to the failed assertion:

```php

if ($customer->id !== $customer->getValue())
```

File: src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php
Comment (mickaelandrieu) [inline src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php:50]: same here: why do we need to be allowed to register after delete to process a deletion?

File: src/Core/Domain/Customer/CommandHandler/EditCustomerHandlerInterface.php
Comment (mickaelandrieu) [inline src/Core/Domain/Customer/CommandHandler/EditCustomerHandlerInterface.php]: interfaces need PHPDoc (even an obvious one :+1: )

**Review by mickaelandrieu — COMMENTED** [review]
some questions, minor comments to address: good job @sarjon

File: src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php
Comment (sarjon) [reply src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php:50]: this is how it works, when deleting customer, admin has an option to choose whether customer can register again or not. Depending on that, it performs soft (wont allow to register again) or hard (will be able to register again) delete.

File: src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php
Comment (sarjon) [reply src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php]: i guess it could be done that way as well, wont change the behavior.

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js:29]: we could, but we dont have a lot of examples for this use case, so i dont want to make "wrong" abstraction.

File: src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php
Comment (mickaelandrieu) [reply src/Adapter/Customer/CommandHandler/BulkDeleteCustomerHandler.php:50]: :+1: cristal clear explanation: thanks

File: src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php
Comment (mickaelandrieu) [reply src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php]: I think it's more readable, go for it please :)

Comment (sarjon) [reply src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php]: and done. :+1:

File: admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js
Comment (matks) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js]: Can we put this "data extraction" processing (lines 58, 59 and 64) in a dedicated function ?

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (matks) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js]: Dont you want to reuse ES6 syntax just like https://github.com/PrestaShop/PrestaShop/pull/11439/files#diff-42affa492f308e9b02a5ccc053f8e330R50 ?

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (matks) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js]: Same here, can we move this process into a dedicated function ?

File: src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php
Comment (matks) [inline src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php:46]: 👍

File: src/Core/Domain/Customer/ValueObject/CustomerDeleteMethod.php
Comment (matks) [inline src/Core/Domain/Customer/ValueObject/CustomerDeleteMethod.php:37]: 👍

**Review by matks — CHANGES_REQUESTED** [review]
Very minor changes needed ;)

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js]: sure, i usually forget to use it. :/

but if you take a look at this
```javascript
$('#' + grid.getId() + '_grid_delete_customers_modal')
```
and this
```javascript
$(`#${grid.getId()}_grid_delete_customers_modal`)
```

for me its much easier (and looks cleaner) to read concatenation thats written in old syntax, especially when theres more than one variable.

File: admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js]: actually, it does not extract data, just creates new input. Either way, this function is really small, and the part you want to extract uses elements defined above, so we would need to pass them around, is it really necessary? :angel:

Comment (matks) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/bulk/customer/delete-customers-bulk-action-extension.js]: In fact what I wanted to achieve is to provide an explanation of what we do here ;)
The function name would have helped 😉 . The code shows what is happening, the function name shows why.

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js
Comment (matks) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/row/customer/delete-customer-row-action-extension.js]: Mmmm I have applied ES6 syntax because I saw it was applied on new JS theme code 🤔 

Maybe @PierreRambaud can share his opinion as he asks for it a lot :) ?

**Comment by matks** [comment]
Status: a few comments to be answered/handled 😉

**Comment by sarjon** [comment]
@matks addressed comments. :+1:

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@sarjon PR approved 😄 however Travis fails because of assets :(
Let's get QA approval before rebase + merge

**Comment by marionf** [comment]
@sarjon 

1) The export doesn't work

capture d ecran_769

2) I have 12 customers, I choose to display 10, but still 12 are displayed

capture d ecran_766

3) In all shop context & default group context, the message "You should select a shop to create a customer" is not displayed

capture d ecran_767

capture d ecran_768

Don't know if 2) & 3) can be solved in this PR or in another one

**Comment by sarjon** [comment]
@marionf 1) and 2) fixed. 3) is already part of another PR. :)

Note that export does not take into account filters as it's being implemented in https://github.com/PrestaShop/PrestaShop/pull/10830

**Comment by marionf** [comment]
It's good, thank you @sarjon :+1:

**Review by matks — DISMISSED** [review]

**Comment by marionf** [comment]
@TristanLDD is it ok for you ? 
Maybe we should add an "actions" label for the last column ?

**Comment by TristanLDD** [comment]
Hello, 

- Indeed, we should add the label "Actions" on top of the last column

- Also, placeholders are missing in the search fields of each column (should be like the catalog page)

- Can you align vertically all the elements of the pagination system:

capture d ecran 2018-12-19 a 11 19 52

Thanks :)

**Comment by sarjon** [comment]
all done.

@LouiseBonnard can you check https://github.com/PrestaShop/PrestaShop/pull/11439/commits/47b87bd513b5c1a457701f3999081a014aa86c0e commit to see if translation keys / domains are okay?

**Comment by LouiseBonnard** [comment]
@sarjon, thanks for pinging me on this one. ;-)

I guess it should be localized in _Admin.Orderscustomers.Help_, since it is about the Customers page. Or even in _Admin.Actions_. Both seem alright for me.

**Comment by matks** [comment]
@sarjon I started the rebase but there are some conflicts that make sense in addition to the usual assets re-build needed. Since both sides make sense, I'll let you handle this one 😉

**Comment by sarjon** [comment]
@matks rebased. :+1:

**Comment by sarjon** [comment]
any ideas why tests fail?

**Comment by matks** [comment]
There was an issue with `develop` that has been solved (see https://github.com/PrestaShop/PrestaShop/pull/12150)
I relaunch tests, let's see if it was the reason of the failure

**Comment by sarjon** [comment]
@matks all good now, merge? ^^

**Comment by matks** [comment]
Yes ! Thanks @sarjon

**Review by matks — APPROVED** [review]

**Pull request merged**
